main.cpp

```diff
@@ -64,151 +63,592 @@ New/This/Pointers/References conclusion
+    *value = *value / *it.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// IntType member functions implementation
+// Primitives
+IntType& IntType::add(int v)
+{
+    std::cout << "(it) " << *value << " + " << v << " = ";
+    *value += v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::subtract(int v)
+{
+    std::cout << "(it) " << *value << " - " << v << " = ";
+    *value -= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::multiply(int v)
+{
+    std::cout << "(it) " << *value << " * " << v << " = ";
+    *value *= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::divide(int v)
+{
+    if (v == 0.0f)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(it) " << *value << " / " << v << " = ";
+    *value /= v;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// IntType member functions implementation
+// FloatType argument
+IntType& IntType::add(const FloatType& ft)
+{
+    std::cout << "(it) " << *value << " + (ft) " << *ft.value << " = ";
+    *value = *value + *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::subtract(const FloatType& ft)
+{
+    std::cout << "(it) " << *value << " - (ft) " << *ft.value << " = ";
+    *value = *value - *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::multiply(const FloatType& ft)
+{
+    std::cout << "(it) " << *value << " * (ft) " << *ft.value << " = ";
+    *value = *value * *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::divide(const FloatType& ft)
+{
+    if (*ft.value == 0)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(it) " << *value << " / (ft) " << *ft.value << " = ";
+    *value = *value / *ft.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+// IntType member functions implementation
+// DoubleType argument
+IntType& IntType::add(const DoubleType& dt)
+{
+    std::cout << "(it) " << *value << " + (dt) " << *dt.value << " = ";
+    *value = *value + *dt.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::subtract(const DoubleType& dt)
+{
+    std::cout << "(it) " << *value << " - (dt) " << *dt.value << " = ";
+    *value = *value - *dt.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::multiply(const DoubleType& dt)
+{
+    std::cout << "(it) " << *value << " * (dt) " << *dt.value << " = ";
+    *value = *value * *dt.value;
+    std::cout << *value << '\n';
+    return *this;
+}
+
+IntType& IntType::divide(const DoubleType& dt)
+{
+    if (*dt.value == 0)
+    {
+        std::cout << "Error: cannot divide by 0!\n";
+    }
+    std::cout << "(it) " << *value << " / (dt) " << *dt.value << " = ";
+    *value = *value / *dt.value;
+    std::cout << *value << '\n';
```

Comment:
Same issue as with FloatType being passed in mentioned above.

Reply:
Same result as above
`(it) 1 / (dt) 0 = -2147483648`